test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/persistence/WorkbenchPersistenceIntegrationTest.java

```diff
@@ -164,11 +164,14 @@ public void testWorkbenchProjectPersistence() {
         assertThat(containersResponse.getResult().getContainers().get(0).getContainerId()).isEqualTo(CONTAINER_ID);
     }
 
-    private void verifyOneServerTemplateWithContainer(String containerId) {
-        ServerTemplateList serverTemplates = kieControllerClient.listServerTemplates();
-        assertThat(serverTemplates.getServerTemplates()).as("Number of server templates differ.").hasSize(1);
+    private ServerTemplate[] verifyOneServerTemplate() {
+        ServerTemplate[] serverTemplates = AwaitilityUtils.untilIsNotEmpty(() -> kieControllerClient.listServerTemplates().getServerTemplates());
+        assertThat(serverTemplates).as("Number of server templates differ.").hasSize(1);
```

Comment:
Right now this method does two things, returns list of server templates and at the same time checking that there is just one template.
Can you move this assertion to verifyOneServerTemplateWithContainer(), possibly renaming this method to getServerTemplates?

Reply:
I did this way because I needed to use within the method scaleToZeroAndToOne. Instead of the changes you suggested, I did:
- The method verifyOneServerTemplate won't return anything: so it will only do one thing as good practice.
- The method verifyOneServerTemplateWithContainer won't call to verifyOneServerTemplate (since it was already called by scaleToZeroAndToOne). To make this method safer, I provided a new method in Awaitibility called untilAsserted, so we can easily assert things using this framework.